test_jsonpath_test.go

```diff
@@ -4165,6 +4185,28 @@ func TestRetrieve_filterLogicalCombination(t *testing.T) {
 				inputJSON:    `[{"a":1},{"a":1.1},{"a":2.9},{"a":3}]`,
 				expectedJSON: `[{"a":1.1},{"a":2.9}]`,
 			},
+			{
+				jsonpath:     `$[?((1==2) && @.a>1)]`,
+				inputJSON:    `[{"a":1},{"a":2},{"a":3}]`,
+				expectedJSON: `[]`,
+				expectedErr:  createErrorMemberNotExist(`[?((1==2) && @.a>1)]`),
```

Comment:
Its unclear why an error is returned in that case. The first term  does not specify any member.

Reply:
This error (ErrorMemberNotExist) will be thrown if the object/array member specified in JSONPath does not exist in the JSON object.

It is in README [* Error Handling](https://github.com/AsaiYusuke/jsonpath#-error-handling).

Reply:
Yes, but its filter. 
That said that applies for all filter, so for that issue the behavior is consistent.

The tests are good.

Reply:
Looking at [json-path-comparison](https://github.com/cburgmer/json-path-comparison), it seems that some implementations may return Not supported error depending on given expression.
So, this PR will only address the Panic issue without changing the current behavior, but it needs to be reviewed to follow something standard as my future ToDo.

Reply:
For that PR: fully agree
I  found the consensus to be less reliable on features that are not supported across the board